tests/functional/d/disable_wrong_import_position.py

```diff
@@ -1,7 +1,10 @@
-"""Checks that disabling 'wrong-import-position' on a statement prevents it from
-invalidating subsequent imports."""
+"""Checks that disabling 'wrong-import-position' only affects the specific line.
+
+A pragma on a non-import statement should not affect subsequent import statements.
+This demonstrates the correct behavior after fixing the bug.
+"""
 # pylint: disable=unused-import
 
 CONSTANT = True  # pylint: disable=wrong-import-position
 
-import sys
+import sys  # [wrong-import-position]
```

Comment:
I think disabling on the line that create the issue instead of the multiple imports that are affected, made sense and was instinctive. I would test for this:

```suggestion
import os
import sys

CONSTANT = False # pylint: disable=wrong-import-position

import time

SOME_OTHER_CONSTANT = True

import logging  # [wrong-import-position]
```

Reply:
I think I understand now - You suggest putting the pragma on the non-import line that breaks the import block, so it suppresses the following imports until the next non-import. Inline disables on the import line should still work. My new changes follow this logic - please let me know if that what you meant.